drivers/adin2111/bm_adin2111.h

```diff
@@ -0,0 +1,19 @@
+#include "adin2111.h"
+#include "network_interface.h"
+#include "util.h"
+
+typedef struct {
+  void *device_handle;
+  NetworkInterfaceCallbacks const *callbacks;
+} Adin2111;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+BmErr adin2111_init(Adin2111 *self);
+NetworkInterface prep_adin2111_netif(Adin2111 *self);
```

Comment:
This is the new driver public interface. All you have to do is call these two functions. See the tests for example usage.